website/guides/running-terraform-in-automation.html.md

```diff
@@ -0,0 +1,279 @@
+to use the `-var` and `-var-file` options on `terraform plan` to specify any
+variable values that would traditionally have been manually-entered under
+interactive usage.
+
+## Plan and Apply on different machines
+
+When running in an orchestration tool, it can be difficult or impossible to
+ensure that the `plan` and `apply` subcommands are run on the same machine,
+in the same directory, with all of the same files present.
+
+Running `plan` and `apply` on different machines requires some additional
+steps to ensure correct behavior. A robust strategy is as follows:
+
+* After `plan` completes, archive the entire working directory, including the
+  `.terraform` subdirectory created during `init`, and save it somewhere
+  where it will be available to the apply step. A common choice is as a
+  "build artifact" within the chosen orchestration tool.
+* Before running `apply`, obtain the archive created in the previous step
+  and extract it _at the same absolute path_. This re-creates everything
+  that was present after plan, avoiding strange issues where local files
+  were created during the plan step.
+
+Terraform currently makes some assumptions which must be accommodated by
+such an automation setup:
+
+* The saved plan file can contain absolute paths to child modules and other
+  data files referred to by configuration. Therefore it is necessary to ensure
+  that the archived configuration is extracted at an identical absolute path.
+  This is most commonly achieved by running Terraform in some sort of isolation,
+  such as a Docker container, where the filesystem layout can be controlled.
+* Terraform assumes that the plan will be applied on the same operating system
+  and CPU architecture as where it was created. For example, this means that
+  it is not possible to create a plan on a Windows computer and then apply it
+  on a Linux server.
+
+## Interactive Approval of Plans
+
+Another challenge with automating the Terraform workflow is the desire for an
+interactive approval step between plan and apply. To implement this robustly,
+it is important to ensure that either only one plan can be outstanding at a
+time or that the two steps are connected such that approving a plan passes
+along  enough information to the apply step to ensure that the correct plan is
+applied, as opposed to some later plan that also exists.
+
+Different orchestration tools address this in different ways, but generally
+this is implemented via a _build pipeline_ feature, where different steps
+can be applied in sequence, with later steps having access to data produced
+by earlier steps.
+
+The recommended approach is to allow only one plan to be outstanding at a
+time. When a plan is applied, any other existing plans that were produced
+against the same state are invalidated, since they must now be recomputed
+relative to the new state. By forcing plans to be approved (or dismissed) in
+sequence, this can be avoided.
+
+## Auto-Approval of Plans
```

Comment:
I see there's more parts like that and I assume we're going to release this guide with 0.10, so I ignore me 🤷‍♂️